plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -1893,6 +1894,56 @@ public void testJsonS3SelectPushdownWithSpecialCharacters()
         }
     }
 
+    @Test
+    public void testS3SelectExperimentalPushdown()
+    {
+        // Demonstrate correctness issues which have resulted in pushdown for TEXTFILE
+        // using CSV support in S3 Select being put behind a separate "experimental" flag.
+        // TODO: https://github.com/trinodb/trino/issues/17775
+        Session usingAppendInserts = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "insert_existing_partitions_behavior", "APPEND")
+                .build();
+        List<String> values = ImmutableList.of(
+                "1, true, 11",
+                "2, true, 22",
+                "3, NULL, NULL",
+                "4, false, 44");
+        Session withS3SelectPushdown = Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "s3_select_pushdown_enabled", "true")
+                .setCatalogSessionProperty("hive", "json_native_reader_enabled", "false")
+                .setCatalogSessionProperty("hive", "text_file_native_reader_enabled", "false")
+                .build();
+        try (TestTable table = new TestTable(
+                sql -> getQueryRunner().execute(usingAppendInserts, sql),
+                "hive.%s.test_s3_select_pushdown_experimental_features".formatted(HIVE_TEST_SCHEMA),
+                "(id INT, bool_t BOOLEAN, int_t INT) WITH (format = 'TEXTFILE')",
+                values)) {
+            assertQuery(withS3SelectPushdown, "SELECT count(*) FROM " + table.getName() + " WHERE int_t IS NULL", "VALUES 0");
+            assertQueryFails(
+                    withS3SelectPushdown,
+                    "SELECT id FROM " + table.getName() + " WHERE bool_t = true",
+                    "S3 returned an error: Error casting:.*");
+        }
+
+        List<String> specialCharacterValues = ImmutableList.of(
+                "1, 'a,comma'",
+                "2, 'a|pipe'",
+                "3, 'an''escaped quote'",
+                "4, 'a~null encoding'");
+        try (TestTable table = new TestTable(
+                sql -> getQueryRunner().execute(usingAppendInserts, sql),
+                "hive.%s.test_s3_select_pushdown_special_characters".formatted(HIVE_TEST_SCHEMA),
+                "(id INT, string_t VARCHAR) WITH (format = 'TEXTFILE', textfile_field_separator=',', textfile_field_separator_escape='|', null_format='~')",
+                specialCharacterValues)) {
+            // These two should return a result, but incorrectly return nothing
+            assertThat(query(withS3SelectPushdown, "SELECT id FROM " + table.getName() + " WHERE string_t ='a,comma'")).returnsEmptyResult();
```

Comment:
Thanks, adding the first bullet exposed a bug in the native textfile reader implementation, submitted a separate issue: https://github.com/trinodb/trino/issues/18215

Reply:
> We also need a blind correctness test ensuring that when S3Select is enabled (but the experimental pushdown is not), results are correct.

Added a separate test class for this, since the experimental flag is not configurable by a session property.